### ISSUE TITLE
Add error parser / normaliser

### ISSUE THREAD

**Issue by sjurgis** [ruslan-kurchenko/sfdc-lax#19]
The resulting error array is quite painful to parse, as a result I've wrote this spaghetti that recursively parses errors into a single array of messages.

It could probably be improved by adding errorType property.
It would be nice to just go `enqueue(...).withErrorParser()` and get simply iterable list of exceptions that you could just `.join()` and print to the screen.

```
getErrors : function(errors) {
   var messages = [];
   var helper = this;
   if(errors){
     var errata = Array.isArray(errors) ? errors : errors.entries;
     if(errata){
       errata.forEach(function(i){
        if(i.message) messages.push(i.message);
        if(i.pageErrors && i.pageErrors.length > 0) messages = [].concat(messages, helper.getErrors(i.pageErrors));
        if(i.fieldErrors) messages = [].concat(messages, helper.getErrors([].concat.apply([], Object.keys(i.fieldErrors).map(function(key) { return i.fieldErrors[key] }))));
       }) 
     }
   }
   return messages;
  }
```

**Comment by ruslan-kurchenko** [comment]
Hey @sjurgis!
Sounds good to me.
What about "plugin/feature" based approach? For example, create hooks in the component behavior where you can embed custom functionality as you mentioned? 
Or, do you think it will be better to "hardcode/add" your function (decorator) directly to the component?

**Comment by ruslan-kurchenko** [comment]
Merged into #21
